lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -275,6 +275,27 @@ def get_cpu_facts(self, collected_facts=None):
                 cpu_facts['processor_vcpus'] = (cpu_facts['processor_threads_per_core'] *
                                                 cpu_facts['processor_count'] * cpu_facts['processor_cores'])
 
+                # if the number of processors available to the module's
+                # thread cannot be determined, the processor count
+                # reported by /proc will be the default:
+                cpu_facts['processor_nproc'] = processor_occurence
+
+                try:
+                    cpu_facts['processor_nproc'] = len(
+                        os.sched_getaffinity(0)
+                    )
+                except AttributeError:
+                    # in Python < 3.3 and MacOS (shouldn't happen here,
+                    # but to be safe) ``sched_getaffinity`` is not available
+                    try:
+                        cmd = self.module.get_bin_path('nproc')
```

Comment:
I moved `get_bin_path` out of `AnsibleModule` and into a standalone function in `lib/ansible/module_utils/common/process.py`. I would recommend using that function directly rather than the method since that is what I would like to move the rest of the code to using eventually.

Reply:
Fixed.

Reply:
I am going to resolve this on my own to raise attention again…